### PR TITLE
Added support for sending streams whose size cannot be computed.

### PR DESCRIPTION
**Pull request by fabianTMC**
by adding chunked transfer-encoding to the form headers.

### CODE REVIEW

**Comment by phal0r** [comment]
Hi @fabianTMC ,

thanks for your PR. Are you aware of any disadvantages of chunked encoding for requests with known content length? I agree that this is good for streams, where the content length is not known, but since this header is applied on every write request, I would like to talk about that before merging.

**Comment by fabianTMC** [comment]
Agreed. The advantage of having a Content-Length header is that the server will be able to make better decisions based on it. I'm not sure if it gets overridden by the Transfer-Encoding header though. It would be better to have a separate API option to enable chunked transfers.

File: lib/weedfs.js [hunk 1]
```diff
@@ -66,10 +66,16 @@ WeedFSClient.prototype = {
...
                     var stream = typeof file[i] === "string" ? fs.createReadStream(file[i]) : null;
                     form.append("file", stream ? stream : file[i]);
-                    var req = form.submit(url, function(err, res) {
+                    var urlParts = url.parse("http://" + (self.usePublicUrl ? finfo.publicUrl : finfo.url) + "/" + finfo.fid + (opts.count == 1 ? "" : "_" + i));
+
+                    var req = form.submit({
+                        host: urlParts.hostname,
+                        path: urlParts.path,
+                        port: urlParts.port,
+                        headers: { 'transfer-encoding': 'chunked' }
```
Comment (phal0r) [inline lib/weedfs.js:77]: I would suggest the same approach as in pull request #7 .

Pass headers via the opts object and set the headers when available in opts.

```
var options = Object.assign({}, url.parse(...));

if (opts.headers) {
            options.headers = opts.headers;
}
form.submit(options)
```

**Review by phal0r — CHANGES_REQUESTED** [review]

**Comment by atroo** [comment]
Support for custom headers added with v1.4.0 (#12).

**Pull request closed without merging**